tests/unit/serialization/test_incremental.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import numpy as np
+import pandas as pd
+
+from arctic._util import are_equals
+from arctic.exceptions import ArcticSerializationException
+from arctic.serialization.incremental import IncrementalDataFrameToRecArraySerializer
+from arctic.serialization.numpy_records import DataFrameSerializer
+from tests.integration.chunkstore.test_utils import create_test_data
+
+from tests.util import get_large_ts
+
+NON_HOMOGENEOUS_DTYPE_PATCH_SIZE_ROWS = 50
+
+onerow_ts = get_large_ts(1)
```

Comment:
Do just turn these into `pytest.fixtures` - as test discovery imports and runs the module you don't want to be doing anything heavyweight on import.

Reply:
sure, hidden these in a lazy initializer, only invoked upon test invocation

Reply:
@jamesblackburn  @bmoscon   I'd like to merge this if no objections (a couple extra commits since the review)

Merging won't have any impact, as the incremental serializer isn't being used in the code, this only adds the implementation.

Once merged, I'll follow up with a subsequent PR which will enable the async internal  serialization/Mongo design, which will actually use the incremental serializer.